### PR TITLE
[IMPROVE] Break concatenation to two messages 

### PR DESCRIPTION
**Pull request by AlexanderKanakis**
### Issue
DF sends intent and activated page fulfillments in the same response.

DF Concat Issue 1

Each response can send multiple fulfillment messages if the flow reaches them.

By concatenation the two messages will be sent in the same "bubble" even though their source is different:
DF Concat Issue 2
 

### Proposed Solution
DS provides diagnostic info about the current request. The diagnostic info contains the intent fulfillment. By cross-referencing the diagnostic info with the response messages, we can separate the intent messages from the page activation massages and concatenate them independently.

DF Concat Issue 3

### Result

- **_Intent_**:

image

- **_Page_**:

image

- **_Livechat_**:

image

### CODE REVIEW

**Comment by ear-dev** [comment]
Test cases:  "Menu" - should concatenate
                     "Invoice" - should not concatenate...  find a test user that has a list

**Comment by ear-dev** [comment]
@AlexanderKanakis this PR is working great, but it concatenates ALL the messages at this point......  at least all the messages that are configured in our current CX.agent.

I talked with Steve and would like to clarify our first iteration of message concatenation strategy.  It will evolve, and include pages, and many other types of transitions eventually, but for now can we do this please:

We would like to concatenate based on `responseType`.  So if two messages in a row share the same `responseType` then we will concatenate.  If not, we do not concatenate.  If three or four messages in a row share the same `responseType` then concatenate them all, until a new message, with a different `responseType` shows up.  

Steve sent this example from his CX logs, so not sure if the data structure you will have to work with is the same on our CX.app.......

```      "responseMessages": [
        {
          "responseType": "ENTRY_PROMPT",
          "text": {
            "text": [
              "Você já utlizou 0 GB do seu pacote de dados mensal. Sua franquia mensal do seu plano Unlimited 50 - 720p - A - V2 é 100 GB e será reiniciado em 08/09/2021."
            ]
          },
          "source": "VIRTUAL_AGENT"
        },
        {
          "source": "VIRTUAL_AGENT",
          "responseType": "HANDLER_PROMPT",
          "text": {
            "text": [
              "Favor, escolha um dos itens do MENU. Você pode retornar as opções digitando 'MENU'."
            ]
          }
        },
        {
          "source": "VIRTUAL_AGENT",
          "responseType": "ENTRY_PROMPT",
          "text": {
            "text": [
              "USO: Confira o seu consumo de dados"
            ]
          }
        },
        {
          "source": "VIRTUAL_AGENT",
          "responseType": "ENTRY_PROMPT",
          "text": {
            "text": [
              "FATURAS: Visualize/Efetue pagamentos"
            ]
          }
        },
        {
          "responseType": "ENTRY_PROMPT",
          "text": {
            "text": [
              "BOLETO: Baixar o Boleto"
            ]
          },
          "source": "VIRTUAL_AGENT"
        }
      ],
```

So that first message with `responseType: HANDLER_PROMPT` sits in it's own bubble.....  i.e. not concatenated with anything.  And the subsequent three messages with `responseType: ENTRY_PROMPT` would all get concatenated. 

Does this make sense?  Let me know what you think......  thanks.

**Comment by ear-dev** [comment]
@AlexanderKanakis Please create a diagram that helps us to understand exactly what you have implemented here.

**Comment by ear-dev** [comment]
@bhardwajaditya @Shailesh351 we are considering merging this PR..... we have tested it and it's working as expected.  Can you just review for style and design please?  Thanks.

File: lib/Dialogflow.ts [hunk 1]
```diff
@@ -413,6 +427,27 @@ class DialogflowClass {
...
+        const executionStep = info['Execution Sequence'][2] ? info['Execution Sequence'][2]['Step 3'] : info['Execution Sequence'][1]['Step 2'];
+
+        if (executionStep) {
+
+            const intentResponses = executionStep.FunctionExecution ? executionStep.FunctionExecution.Responses : null;
+
+            if (intentResponses) {
+                for (const response of intentResponses) {
+                    if (response.text) {
+                        if (response.text.text[0] === text.text[0] && response.responseType === 'HANDLER_PROMPT') {
```
Comment (bhardwajaditya) [inline lib/Dialogflow.ts:441]: We can combine these two if statements

Comment (AlexanderKanakis) [reply lib/Dialogflow.ts]: > We can combine these two if statements

Done

File: lib/Dialogflow.ts [hunk 2]
```diff
@@ -40,19 +39,11 @@ class DialogflowClass {
             };
 
             const queryParams = {
-                timeZone: 'America/Los_Angeles',
-                parameters:  {
+                timeZone: 'America/Los_Angeles', parameters:  {
```
Comment (Shailesh351) [inline lib/Dialogflow.ts:42]: @AlexanderKanakis @ear-dev This is reverting changes of https://github.com/WideChat/Apps.Dialogflow/pull/97. Can you make sure it is necessary?

Comment (ear-dev) [reply lib/Dialogflow.ts:42]: Good catch @Shailesh351 ...........  @AlexanderKanakis can you verify please?  Thanks.

**Pull request merged**